python/ssh-into-emr-cluster/emr_pattern/emr_pattern_stack.py

```diff
@@ -0,0 +1,194 @@
+            subnet_type=ec2.SubnetType.PRIVATE_WITH_EGRESS,
+            cidr_mask=24
+        )
+
+        vpc = ec2.Vpc(self, "infrastructure_vpc",
+            vpc_name="emr_cdk_vpc",
+            nat_gateways=1,
+            cidr= '10.0.0.0/16',
+            gateway_endpoints=
+                {
+                    "S3": 
+                        ec2.GatewayVpcEndpointOptions(service=ec2.GatewayVpcEndpointAwsService.S3)
+                },
+            subnet_configuration=[
+                public_subnet_config,
+                private_subnet_config
+            ],
+        )
+
+        vpc.apply_removal_policy(RemovalPolicy.RETAIN) # manually delete VPC after running 'cdk destroy' to prevent errors
+
+        ec2_security_group = ec2.SecurityGroup(self, "ec2_sg",
+            vpc=vpc,
+            description="sg for ec2 instance that will ssh to emr",
+            security_group_name="ec2_sg_for_emr_pattern",
+            allow_all_outbound=True
+        )
+
+        # Scope down EC2 SG to allow SSH only from your IP if provided in app.py
+        if ssh_origin_ip:
+            ec2_security_group.add_ingress_rule(ec2.Peer.ipv4(f"{ssh_origin_ip}/32"), ec2.Port.tcp(22), 'allow SSH from only one IP')
+
+        # If IP not specified in app.py variables,  allow SSH from any IP (still relatively secure)
+        else:
+            ec2_security_group.add_ingress_rule(ec2.Peer.any_ipv4(), ec2.Port.tcp(22), 'allow SSH from anywhere')
+
+
+        emr_main_node_security_group = ec2.SecurityGroup(self, "emr_main_node_sg",
+            vpc=vpc,
+            description="sg for emr main node",
+            security_group_name="emr_main_node_sg",
+            allow_all_outbound=True
+        )
+
+        emr_main_node_security_group.add_ingress_rule(
+            ec2.Peer.security_group_id(ec2_security_group.security_group_id), 
+            ec2.Port.tcp(22), 
+            'allow traffic to emr main node from EC2 SG to allow for SSH'
+        )
+
+        emr_core_node_security_group = ec2.SecurityGroup(self, "emr_core_node_sg",
+            vpc=vpc,
+            description="sg for emr core nodes",
+            security_group_name="emr_core_node_sg"
+        )
+
+        # EC2 Instance used to SSH into EMR 
+        ec2_instance = ec2.Instance(self, "ec2_instance",
+            vpc=vpc,
+            instance_type=ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.NANO),
+            machine_image=ec2.MachineImage.latest_amazon_linux(),
+            vpc_subnets=ec2.SubnetSelection(
+                subnet_type=ec2.SubnetType.PUBLIC
+            ),
+            security_group=ec2_security_group,
+            key_name="emr_test"
+        )
+
+        # S3 Bucket used for: logging, storing emr scripts, Spark job input data, and Spark job output data
+        s3_bucket = s3.Bucket(
+            self,
+            "s3_bucket",
+            bucket_name=f"emr-cdk-pattern-{Aws.ACCOUNT_ID}-{str(uuid4())[:5]}",
```

Comment:
This is going to cause a bucket to get created on each synth/deploy even you have no changes.

Reply:
@kaizio Thanks for checking through code + finding this bug, good callout! I have made another commit to my fork removing this UUID section for bucket name, as it shouldnt be necessary.